mediacapture-streams/MediaStream-idl.https.html

```diff
@@ -19,33 +19,33 @@ <h1 class="instructions">Description</h1>
+<script src="/common/vendor-prefix.js" data-prefixed-objects='[{"ancestors":["window"], "name":"MediaStream"}]'></script>
 <script>
 var t = async_test("Tests that a MediaStream constructor follows the algorithm set in the spec", {timeout: 10000});
 t.step(function() {
-  navigator.getUserMedia({video: true, audio:true}, t.step_func(function (stream) {
-    var stream1 = new MediaStream();
-    assert_not_equals(stream.id, stream1.id, "Two different MediaStreams have different ids");
-    var stream2 = new MediaStream(stream);
-    assert_not_equals(stream.id, stream2.id, "A MediaStream constructed from another have different ids");
-    var audioTrack1 = stream.getAudioTracks()[0];
-    var videoTrack = stream.getVideoTracks()[0];
-    assert_equals(audioTrack1, stream2.getAudioTracks()[0], "A MediaStream constructed from another share the same audio track");
-    assert_equals(videoTrack, stream2.getVideoTracks()[0], "A MediaStream constructed from another share the same video track");
-    var stream4 = new MediaStream([audioTrack1]);
-    assert_equals(stream4.getTrackById(audioTrack1.id), audioTrack1, "a non-ended track gets added via the MediaStream constructor");
+  navigator.mediaDevices.getUserMedia({video: true, audio:true})
+    .then(t.step_func(function (stream) {
+      var stream1 = new MediaStream();
+      assert_not_equals(stream.id, stream1.id, "Two different MediaStreams have different ids");
+      var stream2 = new MediaStream(stream);
+      assert_not_equals(stream.id, stream2.id, "A MediaStream constructed from another have different ids");
+      var audioTrack1 = stream.getAudioTracks()[0];
+      var videoTrack = stream.getVideoTracks()[0];
+      assert_equals(audioTrack1, stream2.getAudioTracks()[0], "A MediaStream constructed from another share the same audio track");
+      assert_equals(videoTrack, stream2.getVideoTracks()[0], "A MediaStream constructed from another share the same video track");
+      var stream4 = new MediaStream([audioTrack1]);
+      assert_equals(stream4.getTrackById(audioTrack1.id), audioTrack1, "a non-ended track gets added via the MediaStream constructor");
 
-    var audioTrack2 = audioTrack1.clone();
-    audioTrack2.addEventListener("ended", t.step_func(function () {
+      var audioTrack2 = audioTrack1.clone();
+      audioTrack2.stop();
+      assert_equals(audioTrack2.readyState, "ended", "a stopped track is marked ended synchronously");
       var stream3 = new MediaStream([audioTrack2, videoTrack]);
-      assert_equals(stream3.getTrackById(audioTrack2.id), null, "an ended track doesn't get added via the MediaStream constructor");
-      assert_equals(stream3.getTrackById(videoTrack.id), videoTrack, "a non-ended track gets added via the MediaStream constructor even if the previous track was ended");
+      assert_equals(stream3.getTrackById(audioTrack2.id), audioTrack2, "an ended track gets added via the MediaStream constructor");
```

Comment:
Just to verify, does this correspond to changes in the mediacapture-main spec? I checked the spec and the new code tests the correct behavior.

Reply:
I don't recall it ever working that way, but I may have missed it.